### PR TITLE
Improve error message 

### PR DESCRIPTION
**Pull request by abhishekagarwal87**
Few Misc changes
 - We include the resource info in the error message that the user is trying to access. A query can touch many resources at the same time - table, context parameters, etc. it will be useful to fix the query if user knows precisely what they are not allowed to.  
 - If the task completes with zero rows, we now treat it as an ingestion failure. Its almost always a result of misconfiguration and results in confusion. 
 - If time could not be parsed for any row, we get zero intervals in reports. For that too, we return a task fail result with an actionable error message. 

This PR has:
- [ ] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

File: extensions-core/druid-basic-security/src/main/java/org/apache/druid/security/basic/authorization/BasicRoleBasedAuthorizer.java [hunk 1]
```diff
@@ -90,7 +94,11 @@ public Access authorize(AuthenticationResult authenticationResult, Resource reso
     Map<String, BasicAuthorizerRole> roleMap = roleProvider.getRoleMap(name);
 
     if (roleNames.isEmpty()) {
-      return new Access(false);
+      LOG.info("No roles found for authorizer [%s] and user [%s]", name, authenticationResult.getIdentity());
+      return new Access(false, StringUtils.format("You don't have '%s' permission for Resource [%s:%s]",
+          action.name(),
+          resource.getType(),
+          resource.getName()));
```
Comment (LakshSingla) [inline extensions-core/druid-basic-security/src/main/java/org/apache/druid/security/basic/authorization/BasicRoleBasedAuthorizer.java:101]: Is it safe to expose the name of the resource to the end user?
If the error message when a resource doesn't exist versus when a resource exists but the user is not authorized to access it is not the same, might lead to the user knowing which resources are present in the cluster.

File: extensions-core/druid-basic-security/src/main/java/org/apache/druid/security/basic/authorization/BasicRoleBasedAuthorizer.java [hunk 2]
```diff
@@ -106,8 +114,16 @@ public Access authorize(AuthenticationResult authenticationResult, Resource reso
         }
       }
     }
-
-    return new Access(false);
+    LOG.info("User [%s] is not allowed to [%s] the resource [%s:%s]",
```
Comment (LakshSingla) [inline extensions-core/druid-basic-security/src/main/java/org/apache/druid/security/basic/authorization/BasicRoleBasedAuthorizer.java:117]:
```suggestion
    LOG.info("User [%s] does not have [%s] permission for the resource [%s:%s]",
```
nit: Making the log consistent with the error message

**Comment by paul-rogers** [comment]
Because of a recent Travis change that introduced a "silent" merge conflict, you will need to rebase this PR on the latest master to pick up the newest Travis file.

File: extensions-core/druid-basic-security/src/main/java/org/apache/druid/security/basic/authorization/BasicRoleBasedAuthorizer.java (hunk 1 above)
Comment (abhishekagarwal87) [reply extensions-core/druid-basic-security/src/main/java/org/apache/druid/security/basic/authorization/BasicRoleBasedAuthorizer.java:101]: The PR doesn't change the error code in any way. We are appending the information to the error. I think it is safe to expose the name of the resource that they are trying to access.

**Pull request closed without merging**